br/pkg/lightning/errormanager/errormanager_test.go

```diff
@@ -37,45 +37,44 @@ func TestInit(t *testing.T) {
 	require.NoError(t, err)
 
 	cfg := config.NewConfig()
+	cfg.TikvImporter.Backend = config.BackendLocal
 	cfg.TikvImporter.DuplicateResolution = config.DupeResAlgRecord
+	cfg.TikvImporter.OnDuplicate = config.ReplaceOnDup
+	cfg.App.MaxErrorRecords = 100
 	cfg.App.MaxError.Type.Store(10)
 	cfg.App.TaskInfoSchemaName = "lightning_errors"
 
 	em := New(db, cfg, log.L())
-	require.Equal(t, cfg.TikvImporter.DuplicateResolution, em.dupResolution)
+	require.True(t, em.conflictV1Enabled)
+	require.True(t, em.conflictV2Enabled)
```

Comment:
this behavior is confusing that we can enable 2 duplicate conflict method at same time, in which case do we need this?

Reply:
Add a precheck in https://github.com/pingcap/tidb/pull/42647/commits/c5c1efe206a75282e44841476e999d62e87bf671.